packages/world/src/modules/keysintable/KeysInTableModule.sol

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import { StoreSwitch } from "@latticexyz/store/src/StoreSwitch.sol";
+import { ResourceType } from "../core/tables/ResourceType.sol";
+import { Resource } from "../../Types.sol";
+
+import { IBaseWorld } from "../../interfaces/IBaseWorld.sol";
+import { IModule } from "../../interfaces/IModule.sol";
+
+import { WorldContext } from "../../WorldContext.sol";
+import { ResourceSelector } from "../../ResourceSelector.sol";
+
+import { KeysInTableHook } from "./KeysInTableHook.sol";
+import { KeysInTable, KeysInTableTableId } from "./tables/KeysInTable.sol";
+import { UsedKeysIndex, UsedKeysIndexTableId } from "./tables/UsedKeysIndex.sol";
+
+/**
+ * This module deploys a hook that is called when a value is set in the `sourceTableId`
+ * provided in the install methods arguments. The hook keeps track of the keys that are in a given table.
+ * This mapping is stored in a table registered by the module at the `targetTableId` provided in the
+ * install methods arguments.
+ *
+ * Note: if a table with composite keys is used, only the first key is indexed
+ *
+ * Note: this module currently expects to be `delegatecalled` via World.installRootModule.
+ * Support for installing it via `World.installModule` depends on `World.callFrom` being implemented.
+ */
+contract KeysInTableModule is IModule, WorldContext {
+  using ResourceSelector for bytes32;
+
+  // The KeysInTableHook is deployed once and infers the target table id
+  // from the source table id (passed as argument to the hook methods)
+  KeysInTableHook immutable hook = new KeysInTableHook();
+
+  function getName() public pure returns (bytes16) {
+    return bytes16("index");
```

Comment:
same for `KeysWithValue`, but this name should be something more unique than `index`. It's used to track the installed modules in `World.installModule` / `World.installRootModule`, so right now we'd have a clash between `KeysWithValue` and `KeysInTable` if they're called with the same arguments